compiler-rt/lib/scudo/standalone/tests/secondary_test.cpp

```diff
@@ -336,6 +336,31 @@ TEST_F(MapAllocatorCacheTest, CacheOrder) {
     MemMap.unmap();
 }
 
+TEST_F(MapAllocatorCacheTest, CacheOrderWithRelease) {
+  std::vector<scudo::MemMapT> MemMaps;
+  Cache->setOption(scudo::Option::MaxCacheEntriesCount,
+                   CacheConfig::getEntriesArraySize());
+
+  fillCacheWithSameSizeBlocks(MemMaps, CacheConfig::getEntriesArraySize() - 1,
+                              TestAllocSize);
+
+  // Enable releases and perform a deallocation to trigger releases
+  Cache->setOption(scudo::Option::ReleaseInterval, 0);
```

Comment:
Why don't we just do an explicit release?

Reply:
As in a call to `releaseToOs()` or a direct call to `store()`?

Reply:
a call to releaseToOS()

Reply:
The purpose here is to check the ordering is preserved as the committed entries move into the decommitted list, so only calling `releaseToOS()` will not be sufficient to trigger the mechanisms that move committed entries to the decommitted list.

Reply:
Doing a releaseToOS() moves all the entries to the decommited list and when we have to do eviction, it's supposed to pop the decommited list first. If the order is reversed, then we should be able to detect it, right?

Reply:
Oh I see what you're saying. I confused `releaseAndZeroPagesToOS()` with `releaseToOS()`. I agree, that works better. I can add that now.